suite-native/transactions/src/components/TransactionsList/TransactionIcon.tsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+
+import { Box } from '@suite-native/atoms';
+import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
+import { TransactionType } from '@suite-common/wallet-types';
+import { CryptoIcon, CryptoIconName, Icon, IconName } from '@trezor/icons';
+import { colorVariants } from '@trezor/theme';
+
+import { TransactionIconSpinner } from './TransactionIconSpinner';
+
+type TransactionIconProps = {
+    cryptoIconName: CryptoIconName;
+    transactionType: TransactionType;
+    isAnimated?: boolean;
+};
+
+const ICON_SIZE = 48;
+
+const transactionIconMap: Record<TransactionType, IconName> = {
+    recv: 'receive',
+    sent: 'send',
+    joint: 'placeholder',
+    self: 'placeholder',
+    failed: 'placeholder',
+    unknown: 'placeholder',
+};
+
+type TransactionIconStyleProps = {
+    isAnimated: boolean;
+};
+
+const transactionIconStyle = prepareNativeStyle<TransactionIconStyleProps>(
+    (utils, { isAnimated }) => ({
+        width: ICON_SIZE,
+        height: ICON_SIZE,
+        backgroundColor: utils.colors[isAnimated ? 'gray1000' : 'gray100'],
+        borderRadius: utils.borders.radii.round,
+        padding: 14.5,
+    }),
+);
+
+const cryptoIconStyle = prepareNativeStyle<TransactionIconStyleProps>((utils, { isAnimated }) => ({
+    position: 'absolute',
+    right: -2,
+    bottom: -2,
+    padding: 2,
+    backgroundColor: utils.colors[isAnimated ? 'gray1000' : 'gray0'],
+    borderRadius: utils.borders.radii.round,
+}));
+
+export const TransactionIcon = ({
+    cryptoIconName,
+    transactionType,
+    isAnimated = false,
+}: TransactionIconProps) => {
+    const { applyStyle } = useNativeStyles();
+
+    const animatedIconColor = colorVariants.standard.yellow;
+
+    return (
+        <Box>
+            <Box style={applyStyle(transactionIconStyle, { isAnimated })}>
+                <Icon
+                    name={transactionIconMap[transactionType]}
+                    color={isAnimated ? animatedIconColor : 'gray600'}
+                    size="mediumLarge"
+                />
+            </Box>
+            {isAnimated && (
+                <TransactionIconSpinner radius={ICON_SIZE / 2} color={animatedIconColor} />
```

Comment:
I would like the component API more if we had something like IconSpinner or AnimatedIcon, which would wrap normal icon and add this TransactionIconSpinner. Both components are made with skia, so should be possible?

Reply:
It's weird that you need to send different color based on animation to normal icon and then as well render this TransactionIconSpinner

Reply:
Can you give example of API? I am not sure if I am understand.

Reply:
Just a simple composition like this. Maybe it's not possible, that's why I'm asking. But since both components are skia, it should be possible, right?

```
<TransactionIconSpinner>
  <Icon {...props} />
</TransactionIconSpinner>
```

Reply:
What is the benefit of this solution? IMHO it would make the component only more complicated. Actually, I'm not even sure if it's possible to draw one skia `Canvas` inside of another one.

Reply:
Well the benefit is that when you want to use it in render, you just render 1 thing I guess. And it would be easier to reuse if we ever used it with other icon types (highly unlikely though). 

But doesn't matter, since you cannot render canvas inside canvas.